### PR TITLE
Completely remove ssh keys on repxe

### PR DESCRIPTION
**Pull request by pu239ppy**
- Vaulted SSH Keys are now removed as part of re-PXE process 
- Better error reporting on shell out errors

### CODE REVIEW

File: repxe_host.rb [hunk 1]
```diff
@@ -135,19 +130,15 @@ def cobbler_root_password
 
   vault_command.run_command
 
-  unless vault_command.status.success?
-    raise 'Could not retrieve cobbler password!\n' +
-          vault_command.stdout + '\n' +
-          vault_command.stderr
-  end
+  vault_command.invalid! 'Could not retrieve cobbler password!' unless vault_command.status.success?
```
Comment (vt0r) [inline repxe_host.rb]: `vault_command.error!`

File: repxe_host.rb [hunk 2]
```diff
@@ -135,19 +130,15 @@ def cobbler_root_password
...
 
   JSON.parse(vault_command.stdout)['root-password']
 end
 
 def cobbler_sync
   c = Mixlib::ShellOut.new('sudo', 'cobbler', 'sync')
   c.run_command
-  raise 'Failed to sync cobbler' unless c.status.success?
+  c.invalid! 'Failed to sync cobbler' unless c.status.success?
```
Comment (vt0r) [inline repxe_host.rb]: `c.error!`

File: repxe_host.rb [hunk 3]
```diff
@@ -171,9 +170,8 @@ def delete_node_data(entry)
    entry[:hostname]].each do |ssh_name|
     del = Mixlib::ShellOut.new('ssh-keygen', '-R', ssh_name)
     del.run_command
-    unless del.status.success?
-      raise "Failed to delete SSH key for #{ssh_name}: #{del.stderr}"
-    end
+    del.invalid! "Failed to delete SSH key for #{ssh_name}: #{del.stderr}" \
```
Comment (vt0r) [inline repxe_host.rb]: `del.error!`

File: repxe_host.rb [hunk 4]
```diff
@@ -171,9 +170,8 @@ def delete_node_data(entry)
...
+      unless del.status.success?
```
Comment (vt0r) [inline repxe_host.rb]: This line can go away if replacing the above one with just `del.error!`

File: repxe_host.rb [hunk 5]
```diff
@@ -269,11 +267,8 @@ def find_chef_env
 
   env_command.run_command
 
-  unless env_command.status.success?
-    raise 'Could not retrieve Chef environment!\n' +
-          env_command.stdout + '\n' +
-          env_command.stderr
-  end
+  env_command.invalid! 'Could not retrieve Chef environment!' \
```
Comment (vt0r) [inline repxe_host.rb]: `env_command.error!`

File: repxe_host.rb [hunk 6]
```diff
@@ -269,11 +267,8 @@ def find_chef_env
...
+      unless env_command.success?
```
Comment (vt0r) [inline repxe_host.rb]: Adios to this line.

File: repxe_host.rb [hunk 7]
```diff
@@ -308,7 +303,7 @@ def unmount_disks(chef_env, vm_entry)
     if c.status.success?
       puts 'Unmounted ' + disk
     else
-      raise 'Could not unmount ' + disk + ' ' + c.stdout + '\n' + c.stderr
+      c.invalid! 'Could not unmount ' + disk
```
Comment (vt0r) [inline repxe_host.rb]: `c.error!`

File: repxe_host.rb [hunk 8]
```diff
@@ -355,7 +350,7 @@ def kill_chef_client(chef_env, vm_entry)
     c.run_command
   end
   confirm_chef_client_down(chef_env, vm_entry)
-  puts 'Chef client is down'
+  puts 'Chef client is down'
```
Comment (vt0r) [inline repxe_host.rb]: Accidental modification?

File: repxe_host.rb [hunk 9]
```diff
@@ -370,7 +365,7 @@ def start_chef_client(chef_env, vm_entry)
     puts 'Chef client started.'
   else
     puts 'Chef client did not start successfully: ' +
-         c.stdout + '\n' + c.stderr
+      c.format_for_exception
```
Comment (vt0r) [inline repxe_host.rb]: Could replace this whole conditional with `c.error!`

File: repxe_host.rb [hunk 10]
```diff
@@ -386,7 +381,7 @@ def run_chef_client(chef_env, vm_entry, params = ' ')
     puts 'Chef client ran.'
   else
     puts 'Chef client did not run successfully: ' +
-         c.stdout + '\n' + c.stderr
+         c.format_for_exception
```
Comment (vt0r) [inline repxe_host.rb]: Sam here. Unnecessary conditional that avoids using the builtins.

File: repxe_host.rb [hunk 11]
```diff
@@ -418,7 +413,7 @@ def stop_all_services(chef_env, vm_entry)
       puts 'Stopped ' + service
     else
       puts 'Could not stop service ' +
-           service + ' ' + c.stdout + '\n' + c.stderr
+           service + ' ' + c.format_for_exception
```
Comment (vt0r) [inline repxe_host.rb]: and here

File: repxe_host.rb [hunk 12]
```diff
@@ -432,7 +427,7 @@ def shutdown_box(chef_env, vm_entry)
   c.run_command
   if !c.status.success?
     raise 'Could not shut down host ' +
-          vm_entry[:hostname] + '\n' + c.stdout + '\n' + c.stderr
+          vm_entry[:hostname] + '\n' + c.format_for_exception
```
Comment (vt0r) [inline repxe_host.rb]: Finally here.

**Comment by pu239ppy** [comment]
[Mixlib::ShellOut::error!](http://www.rubydoc.info/github/opscode/mixlib-shellout/Mixlib/ShellOut#error!-instance_method) does not take a custom message string, hence why I chose not to use it

**Comment by vt0r** [comment]
*sigh...*

After reviewing the `Mixlib::ShellOut` code and talking it over with Lev, I'm ok with this as is. The uses of `invalid!` are likely the shortest way to achieve what we want (custom message + `error!` behavior). Additionally, we have requirements for the output text format on both successful and unsuccessful attempts in the cases with the conditionals, so this is about the best it can be without writing some external library to do the outputs, which is far out of scope for this PR.

**Comment by cbaenziger** [comment]
@pu239ppy Could you provide a more in-depth description on what this pull request provides? I'm not sure I can grok from your description and the code?

**Pull request merged**